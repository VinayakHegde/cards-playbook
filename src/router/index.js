import Vue from "vue";
import VueRouter from "vue-router";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    redirect: {
      name: "Cards"
    }
  },
  {
    path: "/deck/new",
    name: "Cards",
    meta: {
      header: "Cards"
    },
    component: () => import(/* webpackChunkName: "new" */ "../views/New.vue")
  },
  {
    path: "/deck/:hash",
    name: "orderedpile",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "orderedpile" */ "../views/OrderedPile.vue")
  }
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

export default router;
