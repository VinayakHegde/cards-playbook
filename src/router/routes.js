export default [
  {
    path: "/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/new",
    name: "Home",
    meta: {
      header: "CARDS"
    },
    component: () =>
      import(/* webpackChunkName: "home" */ "../views/home/Home.vue")
  },
  {
    path: "/deck/:hash",
    name: "Pack",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/pack/Pack.vue"),
    props: true
  },
  {
    path: "/error",
    name: "Error",
    meta: {
      header: "404 | Page not found"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/error/Error.vue"),
    props: true
  },
  {
    path: "*",
    redirect: {
      name: "Error"
    }
  }
];
