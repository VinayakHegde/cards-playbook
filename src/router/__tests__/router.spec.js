import router from "../";
import routes from "../routes";

describe("router", () => {
  it("should match the routes provided", () => {
    expect(router.options.routes).toEqual(routes);
  });
  it("should match the snapshot", () => {
    expect(router).toMatchSnapshot();
  });
});
