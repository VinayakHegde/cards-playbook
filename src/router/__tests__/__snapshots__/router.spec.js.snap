// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router should match the snapshot 1`] = `
VueRouter {
  "afterHooks": Array [],
  "app": null,
  "apps": Array [],
  "beforeHooks": Array [],
  "fallback": false,
  "history": HTML5History {
    "base": "",
    "current": Object {
      "fullPath": "/",
      "hash": "",
      "matched": Array [],
      "meta": Object {},
      "name": null,
      "params": Object {},
      "path": "/",
      "query": Object {},
    },
    "errorCbs": Array [],
    "pending": null,
    "ready": false,
    "readyCbs": Array [],
    "readyErrorCbs": Array [],
    "router": [Circular],
  },
  "matcher": Object {
    "addRoutes": [Function],
    "match": [Function],
  },
  "mode": "history",
  "options": Object {
    "base": undefined,
    "mode": "history",
    "routes": Array [
      Object {
        "path": "/",
        "redirect": Object {
          "name": "Home",
        },
      },
      Object {
        "path": "/deck/",
        "redirect": Object {
          "name": "Home",
        },
      },
      Object {
        "component": [Function],
        "meta": Object {
          "header": "CARDS",
        },
        "name": "Home",
        "path": "/deck/new",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "header": "Ordered Pile",
        },
        "name": "Pack",
        "path": "/deck/:hash",
        "props": true,
      },
      Object {
        "component": [Function],
        "meta": Object {
          "header": "404 | Page not found",
        },
        "name": "Error",
        "path": "/error",
        "props": true,
      },
      Object {
        "path": "*",
        "redirect": Object {
          "name": "Error",
        },
      },
    ],
  },
  "resolveHooks": Array [],
}
`;
